.github/workflows/ci-pr-validation.yml

```
@@ -1,110 +1,277 @@
+
 on:
   pull_request:
     types: [opened, synchronize, reopened]
 permissions:
   contents: read
 concurrency:
-  group: pr-${{ github.ref }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
+permissions: {}
```

actionlint (1.7.12)

[error] key "permissions" is duplicated in "workflow" section. previously defined at line:6,col:1

(syntax-check)